--- a/src/def.h
+++ b/src/def.h
@@ -778,8 +778,8 @@
 INT bin_ggt(OP a, OP b, OP c);
 INT binom(OP oben, OP unten, OP d);
 INT binom_small(OP oben, OP unten, OP d);
-INT bit(OP a, INT i);
-INT bit_longint(OP a, INT i);
+INT bit(OP a, INT i) __attribute__((pure));
+INT bit_longint(OP a, INT i) __attribute__((pure));
 INT brauer_char(OP sn, OP prime, OP bc);
 INT bru_comp(OP a, OP c);
 INT bruch_anfang(void);
@@ -1127,10 +1127,10 @@
 INT einsp_cyclotomic(OP a);
 INT einsp_ff(OP a);
 INT einsp_galois(OP g);
-INT einsp_integer(OP a);
+INT einsp_integer(OP a) __attribute__((pure));
 INT einsp_integervector(OP a);
 INT einsp_kranz(OP a);
-INT einsp_longint(OP a);
+INT einsp_longint(OP a) __attribute__((pure));
 INT einsp_matrix(OP a);
 INT einsp_monopoly(OP a);
 INT einsp_permutation(OP a);
@@ -1144,7 +1144,7 @@
 INT elementarp_permutation(OP a, OP b);
 INT elementary_schur_plet(OP om, OP otab, OP ores);
 INT embedding_mod_into_ord(OP prim, OP part, OP f);
-INT empty_listp(OP a);
+INT empty_listp(OP a) __attribute__((pure));
 INT empty_object(char *t);
 INT emptyp(OP a);
 INT en_forme(OP ma);
@@ -1177,8 +1177,8 @@
 INT eval_polynom(OP poly, OP vec, OP res);
 INT eval_polynom_dir_prod(OP a, OP v, OP c);
 INT even(OP a);
-INT even_integer(OP a);
-INT even_longint(OP a);
+INT even_integer(OP a) __attribute__((pure));
+INT even_longint(OP a) __attribute__((pure));
 INT even_partition(OP a);
 INT even_permutation(OP a);
 INT exchange_alphabets(OP a, OP b);
@@ -2198,7 +2198,7 @@
 INT null_galois(OP a,OP b);
 INT null_gr_given_c_d(OP c, OP d, OP gr);
 INT null_object(char *t);
-INT nullp(OP a);
+INT nullp(OP a) __attribute__((pure));
 INT nullp_bitvector(OP bit);
 INT nullp_bruch(OP a);
 INT nullp_cyclo(OP a);
--- a/src/lo.c
+++ b/src/lo.c
@@ -1647,7 +1647,7 @@
     }
 
 
-static INT ganzvergleich(struct longint *x, struct longint *y)
+static INT __attribute__((pure)) ganzvergleich(struct longint *x, struct longint *y)
     {
     struct loc *alocx, *alocy;
     INT    av,lv;
--- a/src/ma.c
+++ b/src/ma.c
@@ -657,7 +657,7 @@
     ENDR("det_imm_matrix");
 }
 
-static INT co_050995(OP a)
+static INT __attribute__((pure)) co_050995(OP a)
 {
     INT i;
     for (i=0; i< S_P_LI(a) ; i+= 2)
--- a/src/mo.c
+++ b/src/mo.c
@@ -238,7 +238,7 @@
  * Rueckgabewerte: Elementnummer m, falls m Minimum ist;
  *                 -1L, falls kein Minimum existiert.
  */
-static INT setmin(TL_BYTE *a)
+static INT __attribute__((pure)) setmin(TL_BYTE *a)
 {
 	TL_BYTE *_a;
 	INT m;
@@ -255,7 +255,7 @@
  * Rueckgabewerte: TRUE, falls a Teilmenge von b ist;
  *                 FALSE, falls a nicht Teilmenge von b ist.
  */
-static INT a_teilmenge_b(TL_BYTE *a, TL_BYTE *b)
+static INT __attribute__((pure)) a_teilmenge_b(TL_BYTE *a, TL_BYTE *b)
 {
 	TL_BYTE *_a,*_b;
 	INT m;
@@ -275,7 +275,7 @@
  * Rueckgabewerte: TRUE, falls a leer ist;
  *                FALSE, falls a nicht leer ist.
  */
-static INT leer(TL_BYTE *a)
+static INT __attribute__((pure)) leer(TL_BYTE *a)
 {
 	INT m;
 
@@ -2154,7 +2154,7 @@
 
 
 /* berechnet mit Hilfe Euklids den ggT zweier Zahlen a und b. */
-static INT _ggT(INT a, INT b)
+static INT __attribute__((const)) _ggT(INT a, INT b)
 {
 	INT x,y,r;
 
--- a/src/zyk.c
+++ b/src/zyk.c
@@ -32,7 +32,7 @@
  */
 
 /* liefert den gcd (greatest common divisor) von a und b zurueck */
-static INT gcd(INT a, INT b)
+static INT __attribute__((const)) gcd(INT a, INT b)
 {
     INT c=5L;
 
@@ -48,7 +48,7 @@
 }
 
 
-static INT eulerfunc(INT n)
+static INT __attribute__((const)) eulerfunc(INT n)
 {
     INT i,h=0L;
 
